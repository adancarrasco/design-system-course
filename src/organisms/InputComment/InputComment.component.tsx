import * as React from 'react'
import styled from '@emotion/styled'
import ButtonInput from '../../molecules/ButtonInput'
import Badge, { IBadgeProps } from '../../atoms/Badge'

export interface IInputCommentProps extends IInputCommentStyledProps {
  isEditMode?: boolean
  isEditing?: boolean
  placeholder: string
  saveText: string
  onSaveClick: () => void
  onChange?: () => void
  onCancelClick?: () => void
  badgeOptions: IBadgeProps
}

const InputComment: React.FC<IInputCommentProps> = ({
  placeholder,
  saveText,
  onSaveClick,
  onChange,
  onCancelClick,
  badgeOptions,
}) => (
  <InputCommentStyled>
    <Badge {...badgeOptions} />
    <ButtonInput
      placeholder={placeholder}
      saveText={saveText}
      onSave={onSaveClick}
      onChange={onChange}
      onCancelClick={onCancelClick}
    />
  </InputCommentStyled>
)

interface IInputCommentStyledProps {
  borderColor?: string
}

const InputCommentStyled = styled.div<IInputCommentStyledProps>`
  display: flex;
  min-width: 340px;
  > span {
    margin-right: 0.5rem;
    align-self: flex-start;
  }
  > div {
    width: calc(100% - 3.7rem);
    align-self: flex-end;
  }
`

export default InputComment
