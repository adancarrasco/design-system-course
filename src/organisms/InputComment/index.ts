export { default } from './InputComment.component'
export type { IInputCommentProps } from './InputComment.component'
