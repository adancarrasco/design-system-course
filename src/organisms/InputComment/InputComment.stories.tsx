import React from 'react'
import { storiesOf } from '@storybook/react'

import InputComment from '.'

storiesOf('Organisms|InputComment', module).add('Default', () => (
  <div style={{ width: '400px' }}>
    <InputComment
      placeholder="Add a more detailed description..."
      saveText="Save"
      onSaveClick={() => console.log('Saving...')}
      onChange={() => console.log('Changing...')}
      onCancelClick={() => console.log('Cancel click')}
      badgeOptions={{ userName: 'Adan Carrasco' }}
    ></InputComment>
  </div>
))
