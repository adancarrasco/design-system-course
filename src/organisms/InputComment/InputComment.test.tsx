import React from 'react'
import { render } from '@testing-library/react'
import { matchers } from 'jest-emotion'
import '@testing-library/jest-dom/extend-expect'

import InputComment, { IInputCommentProps } from '.'

expect.extend(matchers)

describe('InputComment Component', () => {
  const renderComponent = (props: IInputCommentProps) => render(<InputComment {...props} />)
  let props
  beforeEach(() => {
    props = {
      placeholder: 'Add more detailed description...',
      onSaveClick: jest.fn(),
      badgeOptions: {
        userName: 'Adan Carrasco',
      },
    }
  })
  it('matches snapshot and default render', () => {
    const { getByText } = renderComponent(props)
    const comment = getByText('Add more detailed description...')
    const badge = getByText('AC')
    expect(comment.parentElement.parentElement.parentElement).toMatchSnapshot()
    expect(comment).toBeInTheDocument()
    expect(badge).toBeInTheDocument()
  })
})
