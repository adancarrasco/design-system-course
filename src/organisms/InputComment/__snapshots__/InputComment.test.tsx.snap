// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComment Component matches snapshot and default render 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 340px;
}

.emotion-4 > span {
  margin-right: 0.5rem;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.emotion-4 > div {
  width: calc(100% - 3.7rem);
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.emotion-0 {
  width: 2.7rem;
  height: 2.7rem;
  background-image: url("undefined");
  background-color: #ababab;
  background-size: 2.7rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  font-weight: bold;
  border-radius: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
}

.emotion-2 {
  cursor: pointer;
  background-color: #f1f1f1;
  padding: 0.5rem 0.6rem;
  min-height: 40px;
}

.emotion-2:hover {
  background-color: #e4e4e4;
}

.emotion-1 {
  color: #3d3d3d;
}

<div
  class="emotion-4"
>
  <span
    class="emotion-0"
  >
    AC
  </span>
  <div
    class="emotion-3"
  >
    <div
      class="emotion-2"
    >
      <span
        class="emotion-1"
        color="#3d3d3d"
      >
        Add more detailed description...
      </span>
    </div>
  </div>
</div>
`;
